### DISCUSSION TITLE
Does rustdesk support two PCs for P2P connection

### DISCUSSION THREAD

**Discussion by lllAlex2333** [rustdesk/rustdesk#524] [General]
Because my two PCs are far from the relay server, but they are very close to each other. I would like to ask if rustdesk supports P2P connections without using a relay server?

**Comment by rustdesk** [comment]
RustDesk attempts to make direct P2P connection first, relay server will be used if P2P failes.

**Comment by lllAlex2333** [comment]
I'm currently using microsoft rdp which is smooth. But when I'm using rustdesk ,the connection is very slow.  I suspect that the traffic between two PCs is going through a relay server, not a P2P connection

**Comment by rustdesk** [comment]
I guess P2P failed

**Comment by lllAlex2333** [comment]
I would like to ask if the program has a p2p fail log, and I wangt to know what exactly is causing the P2P fail

**Comment by rustdesk** [comment]
You can move mouse to toolbar icon (the green icon, it may be the other icon per your case) as below, you will know it is direct connected or via relay server.

![Screenshot 2022-05-09 at 9 09 08 PM](https://user-images.githubusercontent.com/71636191/167416897-5a420a87-13e2-4e5c-b7c7-95baefaf5b98.png)

**Comment by rustdesk** [comment]
> I would like to ask if the program has a p2p fail log, and I wangt to know what exactly is causing the P2P fail

Sadly, I do not know either, I am not P2P expert either.

**Reply by beyond9thousand** [reply]
you said rustdesk attempts to make a p2p connection first. now you say that you know nothing about p2p, i am a bit confused

**Reply by mon-jai** [reply]
@beyond9thousand "Not an expert" means he knows something about P2P, but is not mastered.

**Reply by beyond9thousand** [reply]
hmm, i got it working eventually. switched to parsec because the connection kept dropping for some reason

**Reply by Ra2-IFV** [reply]
Seems my RustDesk is not trying to do P2P attempts at all. Even in local network it still uses the relay.

**Comment by lllAlex2333** [comment]
my toolbar icon show it works via relay server. Can the  program support the forced use of P2P mode? Because I'm sure my two PCs can connect P2P through Microsoft's RDP。thanks！

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk/issues/113#issuecomment-868121459

**Reply by 857074609** [reply]
How to use RELAY SERVER by default?

**Reply by Ra2-IFV** [reply]
Click on 3 dots, check "Always connect via relay"

**Reply by 857074609** [reply]
> Click on 3 dots, check "Always connect via relay"

I later realized that relaying is actually the default, whereas achieving true P2P is quite difficult.  Thank

**Comment by pedjas** [comment]
Do I have to set port forwarding on router to allow direct connection via internet?

I do not mean to connect to IP address of wan port but if I use ordinary connection on ID, would it try to connect directly?

**Reply by Ra2-IFV** [reply]
If you connect directly (type host in the address bar and click connect) then yes

**Comment by ameaninglessname** [comment]
Yes, it's working.
Recently, I manage to direct connect to another PC on LAN with IPV6.

My setting (on both sides):
<img width="992" height="710" alt="image" src="https://github.com/user-attachments/assets/d862e819-c3e1-4e2a-b386-403dddaf8a4e" />
<img width="963" height="707" alt="image" src="https://github.com/user-attachments/assets/e5b23b24-501d-4ad8-8049-13f4b0da7057" />

Note: if you self-hosted a server like me, make sure your access to its domain is not using the VPN, or you can't get direction connection.

**Reply by flier268** [reply]
And you need to allow rustdesk pass through the firewall
